reference/conanfile/tools/microsoft.rst

```diff
@@ -632,6 +632,32 @@ Example:
         check_min_vs(self, "192")
 
 
+.. _conan_tools_microsoft_msvs_toolset:
+
+msvs_toolset()
+--------------
+
+Available since: `1.59.0 <https://github.com/conan-io/conan/releases/tag/1.59.0>`_
+
+.. code-block:: python
+
+    def msvs_toolset(conanfile)
+
+Returns the corresponding Visual Studio or msvs platform toolset based on the settings of the given ``conanfile``. For instance, it may return ``v143``
+for ``compiler=Visual Studio`` with ``compiler.version=17``. If ``compiler.toolset`` was set in settings, it has a priority and always returned.
+
+Parameters:
+    - ``conanfile``: Always use ``self``, the current recipe
+
+Example:
+
+.. code-block:: python
+
+    from conan.tools.microsoft import msvs_toolset
+
+    def validate(self):
+        msvs_toolset(self)
```

Comment:
this example doesn't use a returned value?

Reply:
updated to use the returned value